domains/DataCollection/Forms/PatientID/index.js

```diff
@@ -23,10 +27,52 @@ import configArray from './config';
 // });
 
 const PatientIDForm = ({ navigation }) => {
+  // similar to componentDidMount and componenetWillUnmount
+  // runs every 10 seconds in the background to get all Async Data
+  useEffect(() => {
+    const interval = setInterval(() => {
+      checkOnlineStatus()
+        .then((isConnected) => {
+          if (isConnected) {
+            getAllData().then((allAsyncData) => {
+              // contains all the available keys
+              const allKeys = allAsyncData.map((a) => a[0]);
+              allKeys.forEach((item) => {
+                if (item.includes('PatientID-')) {
+                  getData(item)
+                    .then((postParams) => {
+                      postObjectsToClass(postParams)
+                        .then(() => {
+                          deleteData(item);
+                          toRoot();
+                        }, () => {
+                        });
+                    });
+                }
+              });
+            });
+          }
+        });
+    }, 10000);
+    return () => {
+      clearInterval(interval);
+    };
+  }, []);
+
   const toRoot = () => {
     navigation.navigate('Root');
   };
 
+  const generateRandomID = () => Math.random().toString(20).substr(2, 12);
```

Comment:
we should create a `utils` folder at the root of the project and export this to form it